cosmos/x/evm/keeper/host.go

```diff
@@ -0,0 +1,138 @@
+	"pkg.berachain.dev/polaris/cosmos/x/evm/plugins/txpool"
+	"pkg.berachain.dev/polaris/cosmos/x/evm/plugins/txpool/mempool"
+	"pkg.berachain.dev/polaris/eth/core"
+	"pkg.berachain.dev/polaris/eth/core/vm"
+	"pkg.berachain.dev/polaris/lib/utils"
+)
+
+// Compile-time interface assertion.
+var _ core.PolarisHostChain = (*Host)(nil)
+
+type Host struct {
+	// The various plugins that are are used to implement `core.PolarisHostChain`.
+	bp  block.Plugin
+	cp  configuration.Plugin
+	gp  gas.Plugin
+	hp  historical.Plugin
+	pp  precompile.Plugin
+	sp  state.Plugin
+	txp txpool.Plugin
+}
+
+// NewHost creates new instances of the plugin host.
+func NewHost(
+	storeKey storetypes.StoreKey,
+	ak state.AccountKeeper,
+	bk state.BankKeeper,
+	authority string,
+	appOpts servertypes.AppOptions,
+	ethTxMempool sdkmempool.Mempool,
+	offChainKv *offchain.Store,
+	rpcProvider evmrpc.Provider,
+) *Host {
+	// We setup the host with some Cosmos standard sauce.
+	h := &Host{}
+
+	// Build the Plugins
+	h.bp = block.NewPlugin(storeKey)
+	h.cp = configuration.NewPlugin(storeKey)
+	h.gp = gas.NewPlugin()
+	h.hp = historical.NewPlugin(h.bp, offChainKv, storeKey)
+	h.txp = txpool.NewPlugin(h.cp, rpcProvider, utils.MustGetAs[*mempool.EthTxPool](ethTxMempool))
+	return h
+}
+
+// Setup sets up the precompile and state plugins with the given precompiles and keepers. It also
+// sets the query context function for the block and state plugins (to support historical queries).
+func (h *Host) Setup(
+	storeKey storetypes.StoreKey,
+	ak state.AccountKeeper,
+	bk state.BankKeeper,
+	precompiles []vm.RegistrablePrecompile,
+	qc func(height int64, prove bool) (sdk.Context, error),
+) {
+	// Setup the precompile and state plugins
+	h.pp = precompile.NewPlugin(precompiles)
+	h.sp = state.NewPlugin(ak, bk, storeKey, h.cp, h.pp)
+
+	// Set the query context function for the block and state plugins
+	h.sp.SetQueryContextFn(qc)
+	h.bp.SetQueryContextFn(qc)
+}
+
+// GetBlockPlugin returns the header plugin.
+func (h *Host) GetBlockPlugin() core.BlockPlugin {
+	return h.bp
+}
+
+// GetConfigurationPlugin returns the configuration plugin.
+func (h *Host) GetConfigurationPlugin() core.ConfigurationPlugin {
+	return h.cp
+}
+
+// GetGasPlugin returns the gas plugin.
+func (h *Host) GetGasPlugin() core.GasPlugin {
+	return h.gp
+}
+
+func (h *Host) GetHistoricalPlugin() core.HistoricalPlugin {
+	return h.hp
+}
+
+// GetPrecompilePlugin returns the precompile plugin.
+func (h *Host) GetPrecompilePlugin() core.PrecompilePlugin {
+	return h.pp
+}
+
+// GetStatePlugin returns the state plugin.
+func (h *Host) GetStatePlugin() core.StatePlugin {
+	return h.sp
+}
+
+// GetTxPoolPlugin returns the txpool plugin.
+func (h *Host) GetTxPoolPlugin() core.TxPoolPlugin {
+	return h.txp
+}
+
+// GetAllPlugins returns all the plugins.
+func (h *Host) GetAllPlugins() []plugins.BaseCosmosPolaris {
+	return []plugins.BaseCosmosPolaris{h.hp, h.cp, h.gp, h.pp, h.sp}
```

Comment:
missing some plugins here (I think this was from before your pr tho), also "hp" should be "bp"